relnotes/error.deprecation.md

```diff
@@ -0,0 +1,8 @@
+### Deprecate aliases in `ExceptionDefinions` that forward to errors
+
+Symbols like `AssertException` or `OutOfMemoryException` are in fact `Error`
+derivatives in D2. During original ocean porting compatibility aliases for
+those were added to `ocean.core.ExceptionDefinions` module to simplify stage 2
+migration. However now that we put most effort into stages 3 and 4, this
+becomes a liability as catching those errors is never correct when D2 build is
+deployed.
```

Comment:
A pointer in how to deal with it would be nice.

Reply:
Pretty tough call to give generic guidelines for this. But I will give it a try :)

Reply:
It might be pretty extreme, but what about instead of deprecating them, making them behave more like D2, like putting a call to `abort()` in the constructor?

Reply:
@leandro-lucarella-sociomantic but this is exactly why we are introducing things like `verify`- because `abort` behavior is not acceptable for some of our apps :)

Reply:
Yes, I know. All I'm saying is if we keep them with an `abort()`, then we are bringing D1 closer to D2. People will still need to move to `verify()` because errors are broken in D2.

Reply:
> then we are bringing D1 closer to D2

But if no one wants that D2 behaviour, what would be the point? :D

Reply:
Less migration efforts at first (of course eventually you have to address it). I'm fine with merging as is BTW, it was just a comment.